src/components/ItemLookup.vue

```diff
@@ -88,71 +100,100 @@ const onSearchInput = ( inputValue: string ) => {
 	debouncedSearchForItems( inputValue );
 };
 
-const onScroll = async () => {
+const onLoadMore = async () => {
 	const searchReults = await props.searchForItems(
 		props.searchInput,
 		searchedOptions.value.length,
 	);
 	searchedOptions.value = [ ...searchedOptions.value, ...searchReults ];
 };
 
-// the remaining setup translates multilingual SearchedItemOptions to monolingual WikitItemOptions
-
-interface WikitMenuItem {
-	label: string;
-	description: string;
-	tag?: string;
-}
-interface WikitItemOption extends WikitMenuItem {
-	value: string;
-}
-
-function searchResultToMonolingualOption( searchResult: SearchedItemOption ): WikitItemOption {
+// the remaining setup translates multilingual SearchedItemOptions to monolingual MenuItemData
+function searchResultToMonolingualOption( searchResult: SearchedItemOption ): MenuItemData {
 	return {
 		label: searchResult.display.label?.value || searchResult.id,
 		description: searchResult.display.description?.value || '',
 		value: searchResult.id,
 	};
 }
 
-const wikitMenuItems = computed( () => {
+const codexMenuItems: ComputedRef<MenuItemData[]> = computed( () => {
 	return menuItems.value.map( searchResultToMonolingualOption );
 } );
 
-const wikitValue = computed( () => {
-	if ( props.value === null ) {
-		return null;
-	}
-	return searchResultToMonolingualOption( props.value );
-} );
-
-const onWikitOptionSelected = ( value: unknown ) => {
-	const wikitOption = value as WikitItemOption | null;
-	const searchOption = menuItems.value.find( ( item ) => item.id === wikitOption?.value );
+const onCodexOptionSelected = ( selectedItem: string | null ) => {
+	const searchOption = menuItems.value.find( ( item ) => item.id === selectedItem );
 	return onOptionSelected( searchOption ?? null );
 };
 
 const messages = useMessages();
+
+const menuConfig = {
+	visibleItemLimit: 6,
+};
+
+const fieldStatus = computed( (): ValidationStatusType => {
+	if ( !props.error ) {
+		return 'default';
+	}
+	return props.error.type;
+} );
+
+const errorMessages = computed( (): ValidationMessages => {
+	if ( props.error ) {
+		if ( props.error.type === 'error' ) {
+			return { error: props.error.message };
+		}
+		if ( props.error.type === 'warning' ) {
+			return { warning: props.error.message };
+		}
+	}
+	return {};
+} );
+
+/**
+ * We want to pass the searchInput property from the parent component
+ * to the child component. The searchInput property comes in read-only
+ * and receives updates from the parent (it is a ref / computed value).
+ *
+ * In the child, we want to pass the property in as read/write, so that
+ * it can be updated by in `v-model:input-value`. To this end, we create
+ * a copy of the property and add a watcher so that updates from the
+ * parent are propagated, and updates from the child are possible.
+ */
+const searchInputCopy = ref( props.searchInput );
+watch( () => props.searchInput, ( newValue ) => {
+	searchInputCopy.value = newValue;
+}, { immediate: true } );
```

Comment:
It looks like you want chained 2-way binding for the `input-value` prop – from the parent to ItemLookup, and then from ItemLookup to the Codex Lookup component.

Instead of using a watcher, I would suggest using the `useModelWrapper` composable that Codex provides ([see here](https://doc.wikimedia.org/codex/latest/composables/demos/use-model-wrapper.html) for info).

First, in the parent component, you want to pass in the `searchInput` prop using v-model:

```
<template>
    <ItemLookup v-model:searchInput="searchInput" />
</template>
```

Then in ItemLookup, you need to set up a custom "model wrapper" (a writable computed prop that will automatically send the appropriate signals to the parent on change events, basically).

```
<script>
import { CdxLookup, useModelWrapper } from '@wikimedia/codex';
// blah blah..

setup( { props, emit } ) {
    const selection = ref( null );
    const searchInputWrapper = useModelWrapper( 
        toRef( props, 'searchInput' ),
        emit,
        'update:searchInput' // this last arg is only needed if your prop is named something other than "modelValue"
    );

    return {
        searchInputWrapper,
        selection
    }
} 
</script>

<template>
    <cdx-lookup
        v-model:selected="selection"
        v-model:input-value="searchInputWrapper"
    />
</template>
```

This should ensure that any changes made here (or from the Lookup component itself) get propagated back to the parent and vice versa.